accepted/0000-missing-command-lookups.md

```diff
@@ -0,0 +1,70 @@
+# Support Yarn style command/script/bin lookups from the CLI
+
+## Summary
+
+Allow for a CLI experience like:
+
+```sh
+npm install typescript
+npm tsc
+```
+
+## Motivation
+
+I have two main motivations:
+
+- Personal, looking at the list of yarn commands I [use in an average month](https://gist.github.com/orta/489dbfa1b7467804955ea7800adf1494#file-last_month_yarn-txt) (1200) roughly 30% (400) of those commands are using the shorthand syntax to run a command installed into `node_modules/.bin` in a project.
+
+- Professional, it'd be great for the TypeScript docs to be able to simply say: `npm install typescript --save-dev' then `npm tsc index.ts` and there to be no ambiguity like our current recommendation for `npx tsc index.ts`.
```

Comment:
if you suggest they install it first, what ambiguity is there in `npx tsc index.js`?

Reply:
I keep getting feedback because people assume "npx = remote", and not "local then remote", I get that they are not right, but I also thought that until I researched what we should recommend.

Reply:
in npm 7, npx will prompt before auto-installing, so as npm 7 propagates, more people hopefully will have their expectations corrected.

Reply:
I also only learned recently that `npx` prefers local resolution. I’d avoided using `npx` from the start because it makes me super nervous to run remote code (I’m also “that guy” who warns people not to pipe to `curl` even if you read the source because it can be detected server side).

Now knowing about local resolution, I’ll use `npx` in projects where I don’t choose the package manager, but only in the `scripts` section where I’m prompted to also verify there’s a local dependency. And even then it makes me super nervous that the dependency will be removed and introduce a silent vulnerability. So nervous that anytime I go to use an `npx` command and I know my git `HEAD` has changed I **try to remember** to check the `package.json`, and I just know someday I’m gonna forget. Not a great UX! (I’m sure some combination of `npm` flags and a `grep` and an `alias` could make this safe for my preferences, but then I’m moving that risk to anyone I forget to warn!)

IMO requiring a local install, whatever the ultimate API, is an important feature for security and repeatability. And those are worth the distinction of a new API, even if TIMTOWTDI.